source/contracts/reporting/Market.sol

```diff
@@ -171,20 +169,44 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
     }
 
     function updateTentativeWinningPayoutDistributionHash(bytes32 _payoutDistributionHash) public returns (bool) {
+        if (_payoutDistributionHash == tentativeWinningPayoutDistributionHash) {
+            return true;
+        }
+
         IReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
         require(address(_reportingToken) != NULL_ADDRESS);
 
         IReportingToken _tentativeWinningReportingToken = reportingTokens[tentativeWinningPayoutDistributionHash];
         if (address(_tentativeWinningReportingToken) == NULL_ADDRESS) {
             tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
-            _tentativeWinningReportingToken = _reportingToken;
+            return true;
         }
-        if (_reportingToken.totalSupply() > _tentativeWinningReportingToken.totalSupply()) {
+
+        int256 _tentativeWinningStake = getPayoutDistributionHashStake(tentativeWinningPayoutDistributionHash, _tentativeWinningReportingToken);
+        int256 _payoutStake = getPayoutDistributionHashStake(_payoutDistributionHash, _reportingToken);
+
+        if (_payoutStake > _tentativeWinningStake) {
+            secondPlaceTentativeWinningPayoutDistributionHash = tentativeWinningPayoutDistributionHash;
             tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
         }
         return true;
     }
 
+    function getPayoutDistributionHashStake(bytes32 _payoutDistributionHash, IReportingToken _reportingToken) private returns (int256) {
```

Comment:
Consider getting `_payoutDistributionHash` from `_reportingToken` rather than passing it in.  Will increase gas costs slightly, but we can address that later if it turns out that this is a hot path that needs gas optimization.  I feel like having a single parameter will be more clear for reading/auditing.